internal/collector/utils.go

```diff
@@ -84,3 +88,32 @@ func kubeAnnotationsToPrometheusAnnotations(annotations map[string]string) ([]st
 func sanitizeLabelName(s string) string {
```

Comment:
It maybe better to add related unit test.

Reply:
@zouyee @brancz ?

Reply:
I think @zouyee meant to transfer the unit tests relevant from what is currently the dependency as well. I agree that would be good.

Reply:
I am okay with doing that. But that would be potentially adding a lot of redundant code since the test coverage is already high with the current test suite. I can add extra units fill the existing gaps. If the redundant tests are okay, then I can proceed with porting over the test cases from k/k to here. Thoughts?

Reply:
I prefer the redundant tests.

Reply:
Sounds good! Thanks for letting me know  :)

Reply:
I will update by PR once I get the time today.

Reply:
Done!